sklearn/decomposition/pca.py

```diff
@@ -176,25 +176,29 @@ class PCA(_BasePCA):
             0 < n_components < min(X.shape)
         randomized :
             run randomized SVD by the method of Halko et al.
+        lobpcg :
+            run lobpcg_svd by LOBPCG of Knyazev 2001
 
         .. versionadded:: 0.18.0
 
     tol : float >= 0, optional (default .0)
-        Tolerance for singular values computed by svd_solver == 'arpack'.
+        Tolerance for singular values computed by svd_solver == 'arpack'
+        For svd_solver == 'lobpcg', tol must be reasonable, not .0!
```

Comment:
That would also be my suggestion, to change `tol=None` by default, as I have already proposed for spectral clustering, see #11968. 

But that should really be a different PR, since it involves all solvers and must be well tested. Finally, in one of the recent edits of lobpcg.py, we fool proof it, replacing 0 or negative `tol` with just the same internal default as for `tol=None` I do not know how other solvers handle `tol=None` or `tol=0`

Reply:
No, it's not a different PR at this point, because we want to maintain backwards compatibility for the other solvers, but provide a sensible default for lobpcg. The defaults for other solvers can change in the future.

Reply:
I don't think "reasonable, not 0!" is a reasonable way to direct users.

Reply:
But in the current PR, tol goes unused by lobpcg; it is only used by arpack